test/no-for-loop.js

```diff
@@ -482,6 +482,121 @@ ruleTester.run('no-for-loop', rule, {
 				const [ a, b ] = element;
 				console.log(a, b, element);
 			}
+		`),
+
+		// Avoid naming collision when using default element name.
+		testCase(outdent`
+			for (let i = 0; i < arr.length; i += 1) {
+				console.log(arr[i]);
+				const element = foo();
+				console.log(element);
+			}
+		`, outdent`
+			for (const element_ of arr) {
+				console.log(element_);
+				const element = foo();
+				console.log(element);
+			}
+		`),
+
+		// Avoid naming collision when using default element name (different scope).
+		testCase(outdent`
+			function element(element_) {
+				for (let i = 0; i < arr.length; i += 1) {
+					console.log(arr[i], element);
+				}
+			}
+		`, outdent`
+			function element(element_) {
+				for (const element__ of arr) {
+					console.log(element__, element);
+				}
+			}
+		`),
+		testCase(outdent`
+			for (let i = 0; i < arr.length; i += 1) {
+				function element__(element) {
+					console.log(arr[i], element);
+				}
+			}
+		`, outdent`
+			for (const element_ of arr) {
+				function element__(element) {
+					console.log(element_, element);
+				}
+			}
+		`),
```

Comment:
This one need fix, we are not checking the childscope.

Reply:
Updated to check child scopes.